rust/crd/src/lib.rs

```diff
@@ -396,7 +420,7 @@ impl SparkApplication {
         let mut submit_conf: BTreeMap<String, String> = BTreeMap::new();
 
         // resource limits, either declared or taken from defaults
-        if let Some(Resources {
+        if let Ok(Resources {
```

Comment:
This will silently ignore fragment validation errors. Maybe something like `let driver_resources = self.driver_resources()?;`

Same goes for the other similar changes below.

Reply:
Done.